src/__tests__/SSEConnection-test.js

```diff
@@ -0,0 +1,199 @@
+import { sources } from "eventsourcemock";
+import puppeteer from "puppeteer";
+import express from "express";
+
+import SSEConnection from "../SSEConnection";
+import ConnectionEvent from "../ConnectionEvent";
+
+const url = "https://mesosphere.com";
+const wait = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+describe("Server Sent Event Connection", () => {
+  describe("API", () => {
+    let connection;
+    beforeEach(() => {
+      connection = new SSEConnection(url);
+    });
+
+    describe("#constructor", () => {
+      it("fails without an url", () => {
+        expect(() => {
+          new SSEConnection();
+        }).toThrowError();
+      });
+    });
+
+    describe("#open", () => {
+      it("opens a connection", () => {
+        const onOpenEventMock = jest.fn();
+        expect(connection.state).toBe(SSEConnection.INIT);
+        connection.addListener(ConnectionEvent.OPEN, onOpenEventMock);
+
+        connection.open();
+        sources[url].emitOpen();
+
+        expect(sources[url].readyState).toBe(1);
```

Comment:
why do you have one test testing three things here, but three different tests below? 🤔 (I like having three tests more :))